src/router.rs

```diff
@@ -258,96 +260,96 @@ impl Error for TrailingSlash {
 #[cfg(test)]
 mod test {
     use super::Router;
-    use iron::{headers, method, status, Request, Response};
+    use iron::{headers, method, Method, StatusCode, Request, Response};
 
     #[test]
     fn test_handle_options_post() {
         let mut router = Router::new();
         router.post("/", |_: &mut Request| {
-            Ok(Response::with((status::Ok, "")))
+            Ok(Response::with((StatusCode::OK, "")))
         }, "");
         let resp = router.handle_options("/");
-        let headers = resp.headers.get::<headers::Allow>().unwrap();
-        let expected = headers::Allow(vec![method::Method::Post]);
-        assert_eq!(&expected, headers);
+        let headers : Vec<method::Method> = resp.headers.get_all(headers::ALLOW).into_iter().map(|s| s.to_str().unwrap().parse().unwrap()).collect();
```

Comment:
This get's quite long. However, I found no other way of converting a `HeaderValue` into a `Method`.

Reply:
Same here...